### PR TITLE
Linear Mixed Effects Models Tutorial

### PR DESCRIPTION
**Pull request by timshell**
Re: Discussion in Gitter.

As before, advice / code review greatly appreciated!

### CODE REVIEW

**Comment by dustinvtran** [comment]
Excellent work. I like the writing. For the model, the `here` in the notebook should point to a link(?). Also it could help to point out that the formula syntax is lme4's and takes from R. For the criticism, it's always nice to add a concluding sentence to explain what a reader should get out of the plot. (Here, that the residuals have Normal error with mean 0.)

We don't keep examples scripts for tutorials anymore, in order to avoid synchronizing too much duplicate code. Keeping two formats (web page + notebooks) is enough so feel free to delete the linear mixed effects model script.

File: docs/tex/tutorials/linear-mixed-effects-models.tex [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+\begin{lstlisting}[language=Python]
+plt.title("Residuals for Prediced Ratings on Test Set")
+plt.xlim(-4, 4)
+plt.ylim(0, 800)
+plt.hist(yhat_vals - y_test, 75)
+plt.show()
+\end{lstlisting}
+
+\includegraphics[width=450px]{/images/linear-mixed-effects-models.png}
+
```
Comment (dustinvtran) [inline docs/tex/tutorials/linear-mixed-effects-models.tex]: remove any blanklines at the end of a file

File: docs/tex/tutorials/linear-mixed-effects-models.tex [hunk 2]
```diff
@@ -0,0 +1,104 @@
+\title{Linear Mixed Effects Models}
+
+\subsection{Linear Mixed Effects Models}
+
+In linear mixed effects models, we wish to model a linear relation between categorical input variables and a continuous output variable. The data is not drawn independently. Oftentimes, we see clusters in subgroups.
```
Comment (dustinvtran) [inline docs/tex/tutorials/linear-mixed-effects-models.tex]: are the inputs necessarily categorical? I think you mean something like "[...] we wish to model a linear relationship for data points with inputs of varying type, categorized into subgroups, and associated to a continuous output".

File: docs/tex/tutorials/linear-mixed-effects-models.tex [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+An interactive version with Jupyter notebook is available
+\href{http://nbviewer.jupyter.org/github/blei-lab/edward/blob/master/docs/notebooks/linear_mixed_effects_models.ipynb}{here},
+or as a script available at
+\href{https://github.com/blei-lab/edward/blob/master/examples/linear_mixed_effects_models.py}
+{\texttt{examples/linear_mixed_effects_models.py}} in the Github repository.
+
+\subsubsection{Data}
+
+We use the \texttt{InstEval} dataset located \href{http://nbviewer.jupyter.org/github/blei-lab/edward/blob/master/docs/notebooks/linear_mixed_effects_models.ipynb}{here}. It's a dataset of instructor evaluation ratings where our inputs include categories such as \texttt{students} and \texttt{departments}, while our observed variable is the actual rating.
+
```
Comment (dustinvtran) [inline docs/tex/tutorials/linear-mixed-effects-models.tex]: add some (or all(?)) of the data loading here

**Comment by timshell** [comment]
Awesome, just made the changes @dustinvtran

File: docs/tex/tutorials/index.tex [hunk 4]
```diff
@@ -36,6 +36,9 @@ \subsection{Tutorials}
 \href{probabilistic-pca}{Probabilistic PCA} \\
 Dimensionality reduction with latent variables.
 
+\href{linear-mixed-effects-models}{Linear mixed effects models} \\
+Linear modeling of fixed and random variables.
```
Comment (dustinvtran) [inline docs/tex/tutorials/index.tex]: typo? "random variables" -> "random effects"

File: docs/notebooks/linear_mixed_effects_models.ipynb [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+    "ed.set_seed(42)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data\n",
+    "\n",
+    "We use the $\\texttt{InstEval}$ dataset located $\\href{https://github.com/blei-lab/edward/blob/master/examples/data/insteval.csv}{here}$. It's a dataset of instructor evaluation ratings where our inputs include categories such as $\\texttt{students}$ and $\\texttt{departments}$, while our observed variable is the actual rating."
```
Comment (dustinvtran) [inline docs/notebooks/linear_mixed_effects_models.ipynb]: the `href` doesn't work in `ipynb`. instead you have to write
```
[here](https://github.com/blei-lab/edward/blob/master/examples/data/insteval.csv)
```

File: docs/tex/tutorials/linear-mixed-effects-models.tex [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+data['deptcodes'] = data['dept'].astype('category').cat.codes
+data['s'] = data['s'] - 1
+
+train = data.sample(frac=0.8)
+test = data.drop(train.index)
+\end{lstlisting}
+
+\subsubsection{Model}
+
+The main reason we use a mixed model rather than regular linear regression is because we cannot make the independence assumption. Rather, our observations come from sets of groups which can different slopes and intercepts.
```
Comment (dustinvtran) [inline docs/tex/tutorials/linear-mixed-effects-models.tex]: > which can different slopes and intercepts.

"which can have varying slopes and intercepts."

File: docs/tex/tutorials/linear-mixed-effects-models.tex [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+For example:
+\begin{itemize}
+\item The observations from a single student are not independent of each other. Rather, some students may systematically give low (or high) lecture ratings.
+\item The observations of a single teachers are not independent from one another. We'd expect good teachers to get generally good ratings and bad teachers to get generally bad ratings.
```
Comment (dustinvtran) [inline docs/tex/tutorials/linear-mixed-effects-models.tex]: > The observations of a single teachers are not independent from one another.

"The observations from a single teacher are not independent of each other."

> We'd expect

"We expect"

File: docs/tex/tutorials/linear-mixed-effects-models.tex [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+\item The observations of a department are not independent from one another. One department may generally have dry material and thus be rated lower than others.
```
Comment (dustinvtran) [inline docs/tex/tutorials/linear-mixed-effects-models.tex]: > The observations of a department are not independent from one another.

"The observations from a single department are not independent of each other."

**Comment by timshell** [comment]
Great stuff, thanks @dustinvtran

**Comment by dustinvtran** [comment]
Thanks again.

**Pull request merged**